src/sentry/static/sentry/app/redux/reducers/projectSearchEntry.js

```diff
@@ -0,0 +1,48 @@
+import {
+  PROJECT_SEARCH_ENTRIES_GET_REQUEST,
+  PROJECT_SEARCH_ENTRIES_GET_SUCCESS,
+  PROJECT_SEARCH_ENTRIES_GET_FAILURE,
+} from '../actions/projectSearchEntry';
+
+// NOTE: We export the initial state in order to use it in tests
+export const initialState = {
+  loading: false,
+  errorMessage: null,
+  projectSearchEntries: [],
+  allVisibleSelected: false,
+  groupBy: 'name',
+  query: 'name',
+  byIds: {}, // The actual substance entries (TODO: have parentById and childById?)
```

Comment:
Here I would have `visibleIds` and `selectedIds` like it is in `app/redux/reducers/substanceSearchEntry.js` then you should be able to simplify the code mentioned earlier.

Reply:
... I guess you removed it since at the moment it's not being used, but I would suggest we keep it and just fill the visible array in the reducer, like in the substance reducer. We will need paging in this view anyways, and probably selection too.